core/src/impl/Kokkos_CPUDiscovery.cpp

```diff
@@ -71,6 +69,8 @@ int processors_per_node() {
 #endif
 #endif
 #ifdef _SC_NPROCESSORS_ONLN
+  int nprocs     = -1;
+  int nprocs_max = -1;
   nprocs = sysconf(_SC_NPROCESSORS_ONLN);
```

Comment:
You can even do
```suggestion
   int const nprocs = sysconf(_SC_NPROCESSORS_ONLN);
```
and similiarly for `nprocs_max`.